### ISSUE TITLE
Custom Blot for inserting link with static content keeps format active

### ISSUE THREAD

**Issue by simonhyttfors** [quilljs/quill#2983]
TL;DR: So in short, I just want to create a custom Blot for inserting a link with a static text i.e. `innerHTML`.

I have a toolbar handler for uploading documents and then inserting a link to the uploaded document, but the problem itself could be reduced to just inserting an anchor element with a static text. My problem now is that my format seems to still be active after the insertion, if I insert a newline, the link duplicates on the next line, if i insert a space and keep typing, then save my text, the link duplicates at the end of the saved text. Here is my implementation:

```javascript
const Inline = Quill.import('blots/inline')

class AttachmentBlot extends Inline {
    static create(value) {
      let node = super.create()

      node.setAttribute('href', value)
      node.setAttribute('target', '_blank')
      node.innerHTML = 'view attachment'

      return node
    }

    static formats(node) {
      return node.getAttribute('href')
    }
  }
AttachmentBlot.blotName = 'attachment'
AttachmentBlot.tagName = 'a'
AttachmentBlot.className = 'attachment'

Quill.register(AttachmentBlot)

// post initalization
quill.getModule('toolbar').addHandler('attachment', () => documentHandler($('#editor-1'), quill))
```

`documentHandler(...)`:

```javascript
const documentHandler = (editor, quill) => {
  // code for choosing document

  // upload function with callback function for returning upload url
  input.onchange = () => uploadDocument(input.files[0], editor, quill, (url) => {
    // trying to inserting the link in the correct format, supposing my problem is either
    // here or in the class declaration
    const range = quill.getSelection(true)
    quill.insertEmbed(range.index, 'attachment', url)
  })
}
```

So in short, I just want to create a custom Blot for inserting a link with a static text i.e. `innerHTML`.

**Comment by simonhyttfors** [comment]
Solved by inheriting from `blots/embed` and overriding `value`-method in subclass:

```javascript
const Embed = Quill.import('blots/embed')

class AttachmentBlot extends Embed {
  static create(value) {
    let node = super.create()

    node.setAttribute('href', value.url)
    node.setAttribute('target', '_blank')
    node.innerText = value.text

    return node
  }

  // let value be hash containing both href and text
  static value(node) {
    return { url: node.getAttribute('href'), text: node.innerText }
  }

  static formats(node) {
    return node.getAttribute('href')
  }
}
AttachmentBlot.blotName = 'attachment'
AttachmentBlot.tagName = 'A'
AttachmentBlot.className = 'attachment'

Quill.register(AttachmentBlot)
```

Inserting format:

```javascript
const range = quill.getSelection(true)
const value = { url: url, text: text }

quill.insertEmbed(range.index, 'attachment', value)
quill.setSelection(range.index + value.text.length)
```

I'm not sure this is exactly what solved my problem, I have mostly been testing different approaches without really fully understanding Quill, but this is the solution I ended up with.
